sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -276,6 +278,26 @@ public String getHostname() {
         }
     }
 
+    @Override
+    public Mono<Map<Symbol, Object>> getRemoteProperties() {
+        if (this.remoteProperties != null) {
+            return Mono.just(this.remoteProperties);
+        }
+        synchronized (this) {
```

Comment:
Can we avoid using `synchronized` here? See `SimpleTokenCache` in core for an example.

Reply:
Changed to use AtomicReference.